components/Elevation/src/UIColor+MaterialElevation.m

```diff
@@ -0,0 +1,63 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "UIColor+MaterialElevation.h"
+
+#import <CoreGraphics/CoreGraphics.h>
+
+#import "MaterialMath.h"
+#import "UIColor+MaterialBlending.h"
+
+@implementation UIColor (MaterialElevation)
+
+- (UIColor *)mdc_resolvedColorWithTraitCollection:(UITraitCollection *)traitCollection
+                                        elevation:(CGFloat)elevation {
+#if defined(__IPHONE_13_0) && (__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_13_0)
+  if (@available(iOS 13.0, *)) {
+    UIColor *resolvedColor = [self resolvedColorWithTraitCollection:traitCollection];
+    if (traitCollection.userInterfaceStyle == UIUserInterfaceStyleDark) {
+      return [resolvedColor mdc_resolvedColorWithElevation:elevation];
+    } else {
+      return resolvedColor;
+    }
+  } else {
+    return [self mdc_resolvedColorWithElevation:elevation];
```

Comment:
I agree this is confusing, which is why I prefer this method to be iOS13+ only. Different users may have different expectation on the output. The code here will not be reached if it is only available for iOS13+ .

Reply:
It can still be reached on iOS 9-12, but the compiler will emit an error/warning. `performSelector` would not prevent it from being executed. Please have it raise an NSException if we're not going to support earlier iOS versions.

Reply:
Addressed in 60205200fea185c2b707b5082a2e04ffa4d94106 to return `self`, in case `performSelector` is called.